api/tests/core/mails/transactional/users/recredit_birthday_anniversary_to_underage_beneficiary_test.py

```diff
@@ -0,0 +1,69 @@
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+import pytest
+
+import pcapi.core.mails.testing as mails_testing
+from pcapi.core.mails.transactional.users.recredit_birthday_anniversary_to_underage_beneficiary import (
+    get_recredit_birthday_anniversary_to_underage_beneficiary_email_data,
+)
+from pcapi.core.mails.transactional.users.recredit_birthday_anniversary_to_underage_beneficiary import (
+    send_recredit_birthday_anniversary_email_to_underage_beneficiary,
+)
+from pcapi.core.payments import models as payments_models
+import pcapi.core.payments.conf as deposit_conf
+from pcapi.core.testing import override_features
+import pcapi.core.users.factories as users_factories
+from pcapi.models import db
+from pcapi.repository import transaction
+
+
+pytestmark = pytest.mark.usefixtures("db_session")
+
+
+class SendinblueSendNewlyEligibleUserEmailTest:
+    @override_features(ENABLE_SENDINBLUE_TRANSACTIONAL_EMAILS=True)
+    def test_send_recredit_birthday_anniversary_email_to_underage_beneficiary(self):
+        # given
+        user = users_factories.UnderageBeneficiaryFactory(
+            dateOfBirth=(datetime.now() - relativedelta(years=16, days=5)), departementCode="93"
+        )
+
+        # when
+        send_recredit_birthday_anniversary_email_to_underage_beneficiary(user)
+
+        # then
+        assert len(mails_testing.outbox) == 1  # test number of emails sent
+        assert mails_testing.outbox[0].sent_data["template"] == {
+            "id_prod": 303,
+            "id_not_prod": 31,
+            "tags": ["anniversaire_16_17_ans"],
+            "use_priority_queue": False,
+        }
+
+    @override_features(ENABLE_SENDINBLUE_TRANSACTIONAL_EMAILS=True)
+    def test_return_correct_recredit_birthday_anniversary_email_to_underage_beneficiary_email_data_total_credit_amount(
+        self,
+    ):
+        # given
+        user = users_factories.UnderageBeneficiaryFactory(
+            dateOfBirth=(datetime.now() - relativedelta(years=16, days=5)), departementCode="93"
+        )
+
+        with transaction():
+            recredit = payments_models.Recredit(
+                deposit=user.deposit,
+                amount=888,
+                recreditType=deposit_conf.RECREDIT_TYPE_AGE_MAPPING[user.age],
+            )
+            recredit.deposit.amount += recredit.amount
+            user.recreditAmountToShow = recredit.amount if recredit.amount > 0 else None
```

Comment:
Je suppose que ce code a été copié-collé de `recredit_underage_users()` où l'on ne sait pas quel est le montant, mais ici tu le définis (888), donc tu peux simplifier : 

```suggestion
            user.recreditAmountToShow = recredit.amount
```

Reply:
Je ne comprends pas bien pourquoi ce test a besoin de créer un `Recredit`. Ce qui compte, j'ai l'impression, c'est le montant du `Deposit` lié au bénéficiaire. Si le test définit ce montant-là, on n'aura pas besoin de ce `Recredit` (qui ajoute de la complexité), je pense, non ?

Reply:
ok simplifié